application/models/FailedEmail.php

```diff
@@ -0,0 +1,224 @@
+     * Please note that you should have this exact method in all your CActiveRecord descendants!
+     * @param string $className active record class name.
+     * @return FailedEmail the static model class
+     */
+    public static function model($className = __CLASS__): FailedEmail
+    {
+        return parent::model($className);
+    }
+
+    public function getColumns()
+    {
+        return [
+            [
+                'id'             => 'id',
+                'class'          => 'CCheckBoxColumn',
+                'selectableRows' => '100',
+            ],
+            [
+                'name'        => 'buttons',
+                'type'        => 'raw',
+                'filter'      => false,
+                'header'      => gT('Action'),
+            ],
+            [
+                'header' => gT('Status'),
+                'name'   => 'status',
+                'value'  => '$data->status',
+                'htmlOptions' => ['class' => 'nowrap']
+            ],
+            [
+                'header' => gT('Created'),
+                'name'   => 'created',
+                'value'  => '$data->created',
+                'filter'      => false,
+                'htmlOptions' => ['class' => 'nowrap']
+            ],
+            [
+                'header' => gT('Updated'),
+                'name'   => 'updated',
+                'value'  => '$data->updated',
+                'filter'      => false,
+                'htmlOptions' => ['class' => 'nowrap']
+            ],
+            [
+                'header' => gT('Email type'),
+                'name'   => 'email_type',
+                'value'  => '$data->email_type',
+                'htmlOptions' => ['class' => 'nowrap']
+            ],
+            [
+                'header' => gT("Recipient"),
+                'name'   => 'recipient',
+                'value'  => '$data->recipient',
+                'htmlOptions' => ['class' => 'nowrap']
+            ],
+            [
+                'header' => gT('Language'),
+                'name'   => 'language',
+                'value'  => '$data->language',
+            ],
+        ];
+    }
+
+    public function getButtons()
+    {
+        $permissions = [
+            'update' => Permission::model()->hasSurveyPermission($this->surveyid, 'responses', 'update'),
+            'delete' => Permission::model()->hasSurveyPermission($this->surveyid, 'responses', 'delete'),
+            'read'   => Permission::model()->hasSurveyPermission($this->surveyid, 'responses', 'read')
+        ];
+        $buttons = App()->getController()->renderPartial('/failedEmail/partials/buttons', [
+            'id' => $this->id,
+            'permissions' => $permissions
+        ], true);
+        return $buttons;
+    }
+
+    /**
+     * @return string
+     * @throws CException
+     * @throws CHttpException
+     */
+    public function getRawMailBody(): string
+    {
+        $mailer = \LimeMailer::getInstance(true);
```

Comment:
This line is not correct, expects int, 0 1 or 2, not boolean; I think you can leave it empty and it would default to 1, right?